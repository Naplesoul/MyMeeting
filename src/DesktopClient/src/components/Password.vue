<template>
  <div class="login-page">
    <v-snackbar
        top
        color="teal lighten-1 white--text"
        timeout="1600"
        light
        v-model="snack">
      {{snackText}}
    </v-snackbar>
    <v-card
        class="teal lighten-5 password-card"
        elevation="10"
    >
      <v-form
          ref="form"
          v-model="valid">
      <v-container>
        <v-row>
          <v-col class="title1 teal--text">
            设 定
          </v-col>
        </v-row>
        <v-row>
          <v-text-field
              v-model="nickname"
              append-icon="mdi-format-title"
              label="昵称"
              :rules="nicknameRules"
              background-color="teal lighten-5"
              required
              color="teal darken-1"
          ></v-text-field>
        </v-row>
        <v-row>
          <v-text-field
              v-model="password"
              append-icon="mdi-dialpad"
              label="密码"
              type="password"
              :rules="passwordRules"
              background-color="teal lighten-5"
              required
              color="teal darken-1"
          ></v-text-field>
        </v-row>
        <v-row>
          <v-text-field
              v-model="password2"
              append-icon="mdi-dialpad"
              label="再次输入密码"
              type="password"
              :rules="passwordRules"
              background-color="teal lighten-5"
              required
              color="teal darken-1"
          ></v-text-field>
        </v-row>
        <v-row>
          <v-col align="center">
            <v-btn
                color="teal accent-4 white--text"
                class="mr-4"
                @click="setPassword"
                :loading="loading"
                :disabled="loading || !valid || password !== password2"
            >
              确 认
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
      </v-form>
    </v-card>
  </div>
</template>

<script>

import axios from "axios";

export default {
  name: "Login",
  components: {
  },
  data(){
    return{
      email : "",
      loading : false,
      nickname: "",
      password: "",
      password2 : "",
      snack : false,
      snackText : "",
      valid : false,
      passwordRules: [
        v => !!v || '请输入密码',
        v => /^[a-zA-Z0-9_]{6,18}$/.test(v) || '密码是六到十八位字母、数字或下划线',
      ],
      nicknameRules: [
        v => v.length >= 2 || '昵称太短',
      ],
    }
  },
  methods:{
    async setPassword(){
      this.loading = true;
      try{
        const response = await axios(
            {
              method : 'post',
              url : this.GLOBAL.baseURL + '/register',
              data : {
                'token' : this.GLOBAL.token,
                'nickname' : this.nickname,
                'password' : this.password
              }
            })
        this.snackText = '设置成功';
        this.snack = true;
        this.loading = false;
        console.log(response);
        setTimeout(()=>{this.$emit('finish');},1000)
      }catch(error){
        if (error.response){
          this.snackText = error.response.data.error;
        }
        else{
          this.snackText = '无法连接至服务器'
        }
        this.snack = true;
        this.loading = false;
      }
    },
  }
}
</script>

<style scoped>
.title1{
  font-size: 36px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 15px;
}
.password-card{
//margin-top: 20%;
  padding: 5% 8%;
//width: 60%;
  margin: calc(20vh - 150px) 10%;
  //margin: auto;
}
</style>