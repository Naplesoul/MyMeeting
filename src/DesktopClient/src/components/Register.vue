<template>
  <div class="login-page">
    <v-snackbar
        top
        color="teal lighten-1 white--text"
        timeout="1600"
        light
        v-model="snack">
      {{snackText}}
    </v-snackbar>
    <v-card
        class="teal lighten-5 register-card"
        elevation="10"
    >
      <v-form
          ref="form"
          v-model="valid">
      <v-container>
        <v-row>
          <v-col class="title1 teal--text">
            注 册
          </v-col>

        </v-row>
        <v-row>
          <v-text-field
              v-model="email"
              append-icon="mdi-email"
              :rules="emailRules"
              label="邮箱"
              background-color="teal lighten-5"
              clearable
              required
              color="teal darken-1"
          ></v-text-field>
        </v-row>
        <v-row>
          <v-col align="center">
            <v-btn
                class="mr-4"
                @click="back"
            >
              返回
            </v-btn>
          </v-col>
          <v-col align="center">
            <v-btn
                color="teal accent-4 white--text"
                class="mr-4"
                @click="sendEmail"
                :loading="loading"
                :disabled="loading ||!valid"
            >
              {{btnText}}
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
      </v-form>
    </v-card>
  </div>
</template>

<script>

import axios from "axios";

export default {
  name: "Login",
  components: {
  },
  data(){
    return{
      email : "",
      btnText : "发送邮件",
      loading : false,
      snack : false,
      snackText : "",
      valid : false,
      emailRules: [
        v => !!v || '请输入邮箱地址',
        v => /.+@.+\..+/.test(v) || '邮箱地址格式错误',
      ],
    }
  },
  methods:{
    back(){
      this.$emit('back');
    },
    async sendEmail(){
      this.loading = true;
      try{
        const response =await axios(
            {
              method : 'post',
              url : this.GLOBAL.baseURL + '/email',
              data : {
                'email' : this.email,
              }
            })
        this.GLOBAL.email = this.email
        this.snackText = '发送成功';
        this.snack = true;
        this.btnText = '再次发送';
        this.loading = false;
        console.log(response);
        setTimeout(()=>{this.$emit('next');},1000)
      }catch(error){
        if (error.response){
          this.snackText = error.response.data.error;
        }
        else{
          this.snackText = '无法连接至服务器'
        }
        this.snack = true;
        this.btnText = '再次发送邮件';
        this.loading = false;
      }
    },
  }
}
</script>

<style scoped>
.title1{
  font-size: 36px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 15px;
}

.register-card{
//margin-top: 20%;
  padding: 5% 8%;
//width: 60%;
  margin: calc(20vh - 80px) 10%;
  //margin: auto;
}
</style>