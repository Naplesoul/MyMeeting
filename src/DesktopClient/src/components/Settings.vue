<template>
  <div class="settings-page">
    <v-snackbar
        top
        color="teal lighten-1 white--text"
        timeout="1600"
        light
        v-model="snack">
      {{snackText}}
    </v-snackbar>
    <v-card
        class="teal lighten-5 settings-card"
        elevation="10"
      >
      <v-container>
        <v-row>
          <v-col class="title1 teal--text">
              用 户
          </v-col>
        </v-row>
        <v-row no-gutters dense>
          <v-col align="center">
            <v-avatar
                size="80px"
                class="teal lighten-4"
            >
              <img
                  alt="Avatar"
                  :loading="loading"
                  :src="GLOBAL.baseURL + GLOBAL.userInfo.portrait"
              >
            </v-avatar>
          </v-col>

          <v-col align="left">
            <p class="nickname">{{GLOBAL.userInfo.nickname}}</p>
            <p class="userid">ID: {{GLOBAL.userInfo.id}}</p>
          </v-col>
        </v-row>
        <v-row dense class="settings-row">
          <v-col align="center" class="title2">
            <v-switch
                v-model="GLOBAL.openMicrophoneWhenEnter"
                label="麦克风默认开启"
                color="teal"
                hide-details
            ></v-switch>
          </v-col>
        </v-row>
        <v-row dense class="settings-row">
          <v-col align="center" class="title2">
            <v-switch
                v-model="GLOBAL.openCameraWhenEnter"
                label="摄像头默认开启"
                color="teal"
                hide-details
            ></v-switch>
          </v-col>
        </v-row>
        <v-row dense>
          <v-col align="center">
            <v-btn
                color="red accent-4 white--text"
                class="mr-4"
                @click="logout"
                :loading="loading"
                :disabled="loading"
            >
              退出登录
            </v-btn>
          </v-col>
          <v-col align="center">
            <v-btn
                class="mr-4 teal white--text"
                @click="back"
                id="enterBtn"
                :loading="loading"
                :disabled="loading"
            >
              会议 >
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-card>
  </div>
</template>

<script>
import axios from "axios";
const FormData = require("form-data");
const fs = require('fs');

export default {
  name: "Settings",
  components: {
  },
  data(){
    return{
      email : "",
      password : "",
      loading: false,
      file: null,
      snack : false,
      snackText : "",
    }
  },
  methods:{
    logout(){
      this.$emit('logout');
    },
    back(){
      this.$emit('back');
    },
    getPortrait(){

    },
    async uploadPortrait(){
      this.loading = true;
      this.file = fs.createReadStream('local/path/image.jpg');
      const formData = new FormData();
      formData.append('file', this.file);
      console.log(this.file);
      try{
        const response =await axios(
            {
              method : 'post',
              url : this.GLOBAL.baseURL + '/portrait?token='+this.GLOBAL.userInfo.token,
              headers:{'Content-Type': 'multipart/form-data'},
              data : formData
            })
        this.GLOBAL.userInfo.portrait = response.data.filename;
        this.snackText = '上传成功';
        this.snack = true;
        this.loading = false;
        console.log(response.data.filename);
      }catch(error){
        if (error.response){
          this.snackText = error.response.data.error;
        }
        else{
          this.snackText = '无法连接至服务器'
        }
        this.snack = true;
        this.loading = false;
      }
    }
  },
}
</script>

<style scoped>
.title1{
  font-size: 36px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 15px;
}
.title2{
  padding: 1% 10%;
}
.settings-card{
  //margin-top: 20%;
  padding: 3% 3%;
  //width: 60%;
  margin: calc(20vh - 130px) 10%;
}
.userid{
  font-family: "Microsoft YaHei UI", serif;
  font-size: 14px;
  text-align: left;
}
.nickname{
  font-family: "Microsoft YaHei UI", serif;
  font-size: 28px;
  text-align: left;
  margin-bottom: 10px;
}
.file-input{
  cursor: default;
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
}
.file-input:hover{
  cursor: pointer;
}
.settings-row{
  margin: 0;
}
</style>
