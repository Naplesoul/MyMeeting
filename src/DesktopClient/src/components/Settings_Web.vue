<template>
  <div class="settings-page">
    <v-snackbar
        top
        color="teal lighten-1 white--text"
        timeout="1600"
        light
        v-model="snack">
      {{snackText}}
    </v-snackbar>
    <div
        class="settings-card"
    >
      <v-container>
        <v-row>
          <v-col class="title1 white--text">
            用 户
          </v-col>
        </v-row>
        <v-row no-gutters dense>
          <v-col align="center">
            <v-avatar
                size="80px"
                class="transparent"
            >
              <img
                  alt="Avatar"
                  :loading="loading"
                  :src="GLOBAL.baseURL + GLOBAL.userInfo.portrait"
              >
            </v-avatar>
          </v-col>

          <v-col align="left">
            <p class="nickname">{{GLOBAL.userInfo.nickname}}</p>
            <p class="userid">ID: {{GLOBAL.userInfo.id}}</p>
          </v-col>
        </v-row>
        <v-row dense class="settings-row">
          <v-col align="center" class="title2">
            <v-switch
                v-model="GLOBAL.openMicrophoneWhenEnter"
                label="麦克风默认开启"
                color="teal"
                hide-details
            ></v-switch>
          </v-col>
        </v-row>
        <v-row dense class="settings-row">
          <v-col align="center" class="title2">
            <v-switch
                v-model="GLOBAL.openCameraWhenEnter"
                label="摄像头默认开启"
                color="teal"
                hide-details
            ></v-switch>
          </v-col>
        </v-row>
        <v-row dense>
          <v-col align="center">
            <v-btn
                color="red accent-4 white--text"
                class="mr-4"
                @click="logout"
                :loading="loading"
                :disabled="loading"
            >
              退出登录
            </v-btn>
          </v-col>
          <v-col align="center">
            <v-btn
                class="mr-4 white"
                @click="back"
                :loading="loading"
                :disabled="loading"
            >
              会议 >
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </div>
  </div>
</template>

<script>
import axios from "axios";
const FormData = require("form-data");
const fs = require('fs');

export default {
  name: "Settings",
  components: {
  },
  data(){
    return{
      email : "",
      password : "",
      loading: false,
      file: null,
      snack : false,
      snackText : "",
    }
  },
  methods:{
    logout(){
      this.$emit('logout');
    },
    back(){
      this.$emit('back');
    },
    getPortrait(){

    },
    async uploadPortrait(){
      this.loading = true;
      this.file = fs.createReadStream('local/path/image.jpg');
      const formData = new FormData();
      formData.append('file', this.file);
      console.log(this.file);
      try{
        const response =await axios(
            {
              method : 'post',
              url : this.GLOBAL.baseURL + '/portrait?token='+this.GLOBAL.userInfo.token,
              headers:{'Content-Type': 'multipart/form-data'},
              data : formData
            })
        this.GLOBAL.userInfo.portrait = response.data.filename;
        this.snackText = '上传成功';
        this.snack = true;
        this.loading = false;
        console.log(response.data.filename);
      }catch(error){
        if (error.response){
          this.snackText = error.response.data.error;
        }
        else{
          this.snackText = '无法连接至服务器'
        }
        this.snack = true;
        this.loading = false;
      }
    }
  },
}
</script>

<style scoped>
.title1{
  font-size: 36px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 15px;
}
.title2{
  padding: 1% 10%;
  color: white;
}
.settings-card{
  padding: 0 8%;
  width: calc(300px + 10vw);
  position: absolute;
  z-index: 2;
  bottom: calc(60vh - 220px);
  left: calc(45vw - 150px);
  background-position-y: 20px;
  background-image: linear-gradient(to bottom, #26A69A00, #26A69Add);
  border-bottom: 5px solid #A7FFEB;
}
.userid{
  font-family: "Microsoft YaHei UI", serif;
  font-size: 14px;
  text-align: left;
  color: white;
}
.nickname{
  font-family: "Microsoft YaHei UI", serif;
  font-size: 28px;
  text-align: left;
  margin-bottom: 10px;
  color: white;
}
.file-input{
  cursor: default;
  position: absolute;
  opacity: 0;
  top: 0;
  left: 0;
}
.file-input:hover{
  cursor: pointer;
}
.settings-row{
  margin: 0;
}
</style>