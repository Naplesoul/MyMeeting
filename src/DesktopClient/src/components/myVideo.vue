<template>
  <video :id="myId" :autoplay="srcObject"></video>
</template>

<script>
  export default {
    name: "myVideo",
    mounted() {
      this.videoElement = document.getElementById(this.myId)
      this.videoElement.style.transform = this.mirror ? "rotateY(180deg)" : ""
      this.videoElement.srcObject = this.srcObject;
    },
    props: ["srcObject", "mirror", "myId"],
    watch: {
      srcObject(newValue) {
          this.videoElement.srcObject = newValue
      },
      mirror(newValue) {
        this.videoElement.style.transform = newValue ? "rotateY(180deg)" : ""
        this.videoElement.srcObject = this.srcObject
      }
    }
  }
</script>

<style scoped>

</style>
