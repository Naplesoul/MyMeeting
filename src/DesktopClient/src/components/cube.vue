<template>
  <div :class="['td-cube', {'td-cube-main': !account}]">
    <ul :class="[{'page1': currentPage === 'Login',
         'page2': currentPage === 'Register',
         'page3': currentPage === 'Verify',
         'page4': currentPage === 'Password',
         'page5': currentPage === 'Settings',
         'page6': !account, 'rot':rot}, 'ul-page']">
      <li :class="[{'main':!account}, 'front']">
        <transition name="cols-transition">
          <login @register="currentPage='Register'" @login="currentPage='Settings'" v-if="!rot"></login></transition>
      </li>
      <li :class="[{'main':!account}, 'back']">
        <transition name="cols-transition">
          <Register @back="currentPage='Login'" @next="currentPage='Verify'" v-if="!rot"></Register></transition>
      </li>
      <li :class="[{'main':!account}, 'top']">
        <transition name="cols-transition">
          <Verify @back="currentPage='Register'" @ok="currentPage='Password'" v-if="!rot"></Verify></transition>
      </li>
      <li  :class="[{'main':!account}, 'bottom']">
        <transition name="cols-transition">
          <Password @finish="currentPage='Login'" v-if="!rot"></Password></transition>
      </li>
      <li  :class="[{'main':!account}, 'left']"></li>
      <li  :class="[{'main':!account}, 'right']">
        <transition name="cols-transition">
          <Settings id="setting-page" @logout="currentPage='Login'" @back="backToMain" v-if="currentPage==='Settings' && !rot"></Settings></transition>
      </li>
    </ul>
    <div v-if="rot">
      <v-icon class="hint" @click="openSettings" size="40px">mdi-cogs</v-icon>
    </div>
  </div>
</template>

<script>
import Login from './Login'
import Register from "./Register";
import Verify from "./Verify";
import Password from "./Password";
import Settings from "./Settings"

export default {
  components:{
    Login,
    Register,
    Verify,
    Password,
    Settings,
  },

  data() {
    return {
      pages : ['Login', 'Register', 'Verify', 'Password', 'Settings'],
      currentPage : 'Login',
      rot : false,
    };
  },

  props:{
    account : Boolean,
  },

  methods: {
    backToMain(){
      this.$emit('back');
      setTimeout(()=>{
        this.rot = true;
      }, 1000)
    },
    openSettings(){
      if (this.rot){
        this.rot = false;
        setTimeout(()=>{
          this.$emit('settings');
        }, 200);
      }
    }
  },

  created() {
  }
};
</script>
<style lang="scss" scoped>
$width: 50vh;
$height: 50vh;
$length: 50vh;
.td-cube {
  width: $width;
  height: $height;
  perspective: 75vh;
  transition: 0.5s ease-out;
  position: absolute;
  top: 23vh;
  left: calc(50vw - 25vh);
}
.td-cube.td-cube-main {
  transition: 0.5s ease-out;
  top: -20vh;
  left: calc(100vw - 30vh);
}
.hint{
  text-align: center;
  position: relative;
  top: -27vh;
  left: 23vh;
  color: white;
  transition: 0.15s ease-in-out;
}
.hint:hover{
  cursor: pointer;
  transform: scale(1.2);
  transition: 0.15s ease-in-out;
}
li {
    position: absolute;
    width: $width;
    height: $height;
    overflow: hidden;
}
.ul-page{
  display: flex;
  width: 100%;
  height: 100%;
  transform-style: preserve-3d;
  transition: 0.5s ease-in-out;
  padding: 0;
}
.ul-page.rot {
  transition: 0.5s ease-in-out;
  animation: rotate 12s infinite linear;
}
.ul-page.page1{
  transform: rotateY(0deg);
  transition: 0.8s ease-in-out;
}
.ul-page.page2{
  transform: rotateY(180deg);
  transition: 0.8s ease-in-out;
}
.ul-page.page3{
  transform: rotateX(-90deg) rotateY(0);
  transition: 0.8s ease-in-out;
}
.ul-page.page4{
  transform: rotateX(90deg) rotateY(0);
  transition: 0.8s ease-in-out;
}
.ul-page.page5{
  transform: rotateX(0) rotateY(-90deg);
  transition: 0.8s ease-in-out;
}
.ul-page.page6{
  transition: 1s ease-out;
  //animation: rotate 8s infinite linear;
  transform: rotateY(0) rotateX(35deg) rotateZ(45deg) scaleX(0.1) scaleY(0.1) scaleZ(0.1);
}
.front {
  background-color: #10222077;
  transform: translateZ($length / 2);
}
.front.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}
.back {
  background-color: #10222077;
  transform: rotateY(180deg) translateZ($length / 2);
}
.back.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}
.top {
  background-color: #10222077;
  transform: rotateX(90deg) translateZ($height / 2);
}
.top.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}
.bottom {
  background-color: #10222077;
  transform: rotateX(-90deg) translateZ($height / 2);
}
.bottom.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}
.left {
  background-color: #10222077;
  transform: rotateY(-90deg) translateZ($width / 2);
}
.left.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}
.right {
  background-color: #10222077;
  transform: rotateY(90deg) translateZ($width / 2);
}
.right.main {
  background-image: linear-gradient(to bottom left, #00695C55, #00897Bff);
  transition: 0.7s ease-in-out;
  border: 20px solid #80CBC4;
}

.cols-transition-enter, .cols-transition-leave-to{
  opacity: 0;
}
.cols-transition-leave, .cols-transition-enter-to{
  opacity: 1;
}
.cols-transition-enter-active, .cols-transition-leave-active{
  transition: all 0.7s ease-in-out;
}

@keyframes rotate {
 from {
   transform: rotateY(0) rotateX(35deg) rotateZ(45deg) scaleX(0.1) scaleY(0.1) scaleZ(0.1);
 }
 to {
   transform: rotateY(-1turn) rotateX(35deg) rotateZ(45deg) scaleX(0.1) scaleY(0.1) scaleZ(0.1);
 }
}
</style>
