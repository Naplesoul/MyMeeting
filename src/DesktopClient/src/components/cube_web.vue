<template>
  <div :class="['td-cube', {'td-cube-main': !account}]">
    <transition name="cols-transition" mode="out-in">
      <login @register="currentPage='Register'" @login="currentPage='Settings'" v-if="currentPage === 'Login'"></login>
      <Register @back="currentPage='Login'" @next="currentPage='Verify'" v-if="currentPage === 'Register'"></Register>
      <Verify @back="currentPage='Register'" @ok="currentPage='Password'" v-if="currentPage === 'Verify'"></Verify>
      <Password @finish="currentPage='Login'" v-if="currentPage === 'Password'"></Password>
      <Settings @logout="currentPage='Login'" @back="backToMain" v-if="currentPage==='Settings' && account"></Settings>
    </transition>
    <div class="fake-page" v-if="account"></div>

    <div v-if="!account">
      <v-icon class="hint" @click="openSettings" size="40px">mdi-cogs</v-icon>
    </div>
  </div>
</template>

<script>
import Login from './Login_Web'
import Register from "./Register_Web";
import Verify from "./Verify_Web";
import Password from "./Password_Web";
import Settings from "./Settings_Web"

export default {
  components:{
    Login,
    Register,
    Verify,
    Password,
    Settings,
  },

  data() {
    return {
      pages : ['Login', 'Register', 'Verify', 'Password', 'Settings'],
      currentPage : 'Login',
      rot : false,
    };
  },

  props:{
    account : Boolean,
  },

  methods: {
    backToMain(){
      this.$emit('back');
      setTimeout(()=>{
        this.rot = true;
      }, 1000)
    },
    openSettings(){
      if (this.rot){
        this.rot = false;
        setTimeout(()=>{
          this.$emit('settings');
        }, 200);
      }
    }
  },

  created() {
  }
};
</script>
<style lang="scss" scoped>

.td-cube {
  width: 100vh;
  height: 100vh;
  transition: 0.5s ease-out;
  position: absolute;
  left: 0;
  top: 0;
}
.td-cube.td-cube-main {
  transition: 0.5s ease-out;
  top: 0;
  left: 100vw;
}
.hint{
  text-align: center;
  position: relative;
  top: 10px;
  left: -50px;
  color: white;
  transition: 0.15s ease-in-out;
}
.hint:hover{
  cursor: pointer;
  transform: scale(1.2);
  transition: 0.15s ease-in-out;
}

.cols-transition-enter, .cols-transition-leave-to{
  opacity: 0;
}
.cols-transition-leave, .cols-transition-enter-to{
  opacity: 1;
}
.cols-transition-enter-active, .cols-transition-leave-active{
  transition: all 0.3s ease-in-out;
}

.fake-page{
  width: calc(312px + 10vw);
  height: 12px;
  position: absolute;
  bottom: calc(60vh - 220px);
  z-index: 1;
  left: calc(45vw - 156px);
  border: 6px solid #A7FFEB;
}
</style>