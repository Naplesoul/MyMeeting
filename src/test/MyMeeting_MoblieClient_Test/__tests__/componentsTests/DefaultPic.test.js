import { shallow } from 'enzyme';
import 'jsdom-global/register';
import React from 'react';
import { DefaultPic } from '../../src/components/DefaultPic';
import toJson from 'enzyme-to-json';

//DefaultPic, MyPic, DefaultWithAudioPic are similar components, so only test DefaultPic
describe('default picture of peer', () => {
    const normalStyle = {
        width: 300,
        height: 800,
    }

    test('with no avatar', () => {
        const wrapper = shallow(<DefaultPic style={normalStyle} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('with avatar', () => {
        const wrapper = shallow(<DefaultPic style={normalStyle} imgSrc={'testUri.jpg'} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('given width smaller than height', () => {
        const testStyle = {
            width: 200,
            height: 300,
        }
        const wrapper = shallow(<DefaultPic style={testStyle} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})