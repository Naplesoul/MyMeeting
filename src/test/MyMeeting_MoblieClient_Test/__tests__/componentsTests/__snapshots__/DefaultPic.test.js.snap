// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default picture of peer given width smaller than height 1`] = `
<View
  style={
    Array [
      Object {
        "height": 300,
        "width": 200,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#171717",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={Object {}}
    style={
      Object {
        "height": 66.66666666666667,
        "width": 66.66666666666667,
      }
    }
  />
</View>
`;

exports[`default picture of peer with avatar 1`] = `
<View
  style={
    Array [
      Object {
        "height": 800,
        "width": 300,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#171717",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "testUri.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;

exports[`default picture of peer with no avatar 1`] = `
<View
  style={
    Array [
      Object {
        "height": 800,
        "width": 300,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#171717",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={Object {}}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;
